// @aws-sdk/client-kinesis-video/dist-es/commands/ListEdgeAgentConfigurationsCommand.js
import { getEndpointPlugin } from "@smithy/middleware-endpoint";
import { getSerdePlugin } from "@smithy/middleware-serde";
import { Command as $Command } from "@smithy/smithy-client";
import { SMITHY_CONTEXT_KEY, } from "@smithy/types";
import { ListEdgeAgentConfigurationsOutputFilterSensitiveLog, } from "../models/models_0";
import { de_ListEdgeAgentConfigurationsCommand, se_ListEdgeAgentConfigurationsCommand, } from "../protocols/Aws_restJson1";
export { $Command };
export class ListEdgeAgentConfigurationsCommand extends $Command {
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ListEdgeAgentConfigurationsCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "KinesisVideoClient";
        const commandName = "ListEdgeAgentConfigurationsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: (_) => _,
            outputFilterSensitiveLog: ListEdgeAgentConfigurationsOutputFilterSensitiveLog,
            [SMITHY_CONTEXT_KEY]: {
                service: "KinesisVideo_20170930",
                operation: "ListEdgeAgentConfigurations",
            },
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return se_ListEdgeAgentConfigurationsCommand(input, context);
    }
    deserialize(output, context) {
        return de_ListEdgeAgentConfigurationsCommand(output, context);
    }
}
